homeassistant/components/homekit_controller/number.py

```diff
@@ -152,3 +159,66 @@ async def async_set_value(self, value: float):
                 self._char.type: value,
             }
         )
+
+
+class HomeKitEcobeeFanModeNumber(CharacteristicEntity, NumberEntity):
+    """Representation of a Number control for Ecobee Fan Mode request."""
+
+    _attr_name = "Fan Mode"
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity is tracking."""
+        return [self._char.type]
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        prefix = ""
+        if name := super().name:
+            prefix = name
+        return f"{prefix} {self._attr_name}"
+
+    @property
+    def min_value(self) -> float:
+        """Return the minimum value."""
+        return self._char.minValue
+
+    @property
+    def max_value(self) -> float:
+        """Return the maximum value."""
+        return self._char.maxValue
+
+    @property
+    def step(self) -> float:
+        """Return the increment/decrement step."""
+        return self._char.minStep
+
+    @property
+    def value(self) -> float:
+        """Return the current characteristic value."""
+        return self._char.value
+
+    async def async_set_value(self, value: float):
+        """Set the characteristic to this value."""
+        if value > self.min_value:
+            other_value = value - 1
+        else:
+            other_value = self.min_value + 1
+
+        if value != other_value:
```

Comment:
Ditto for a comment about what we are doing here.

Reply:
added comment